src/ann_solo/reader.py

```diff
@@ -186,39 +229,60 @@ def get_spectrum(self, spec_id: int, process_peaks: bool = False)\
         Returns
         -------
         Spectrum
-            The spectrum from the spectral library file with the specified
+            The spectrum from the spectral library store with the specified
             identifier.
         """
-        spectrum = self._parser.read_spectrum(
-            self.spec_info['offset'][spec_id])[0]
+        spectrum = self._spectra_library_store.read_spectrum_from_library(
+                                                spec_id)
         spectrum.is_processed = False
         if process_peaks:
-            process_spectrum(spectrum, True)
+            return process_spectrum(spectrum, True)
 
         return spectrum
 
-    def get_all_spectra(self) -> Iterator[Tuple[MsmsSpectrum, int]]:
-        """
-        Generates all spectra from the spectral library file.
 
-        For each individual spectrum a tuple consisting of the spectrum and
-        some additional information as a nested tuple (containing on the type
-        of spectral library file) are returned.
+    def read_all_spectra(self) -> Iterator[MsmsSpectrum]:
+        """
+        Traverse all spectra from the spectral library store.
 
         Returns
         -------
         Iterator[Tuple[Spectrum, int]]
-            An iterator of all spectra along with their offset in the spectral
-            library file.
+            An iterator of all spectra in the spectral library hdf5 store.
         """
-        self._parser.seek_first_spectrum()
-        try:
-            while True:
-                spectrum, offset = self._parser.read_spectrum()
-                spectrum.is_processed = False
-                yield spectrum, offset
-        except StopIteration:
-            return
+
+        for spec_id in self._spectra_library_store.get_all_spectra_ids():
+            yield self.read_spectrum(spec_id)
+
+
+    def read_library_file(self) -> Iterator[MsmsSpectrum]:
+        """
+        Read all spectra from the splib library file.
+
+        Returns
+        -------
+        Iterator[Spectrum]
+            An iterator of spectra in the given library file.
+        """
+        verify_extension(['.splib', '.sptxt', '.mgf'], self._filename)
+
+        _, ext = os.path.splitext(os.path.basename(self._filename))
```

Comment:
**suggestion:** Rather than splitting off the extension every time here, it seems more logical to do this in the constructor and store the result in an internal variable.